src/watchers/indexable-author.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Author watcher to save the meta data to an Indexable.
+ *
+ * @package Yoast\YoastSEO\Watchers
+ */
+
+namespace Yoast\YoastSEO\Watchers;
+
+use Yoast\YoastSEO\Exceptions\No_Indexable_Found;
+use Yoast\YoastSEO\Formatters\Indexable_Author as Indexable_Author_Formatter;
+use Yoast\YoastSEO\WordPress\Integration;
+use Yoast\YoastSEO\Models\Indexable;
+
+/**
+ * Watches an Author to save the meta information when updated.
+ */
+class Indexable_Author implements Integration {
+
+	/**
+	 * Registers all hooks to WordPress.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'profile_update', array( $this, 'save_meta' ), PHP_INT_MAX, 2 );
+		\add_action( 'deleted_user', array( $this, 'delete_meta' ) );
+	}
+
+	/**
+	 * Deletes user meta.
+	 *
+	 * @param int $user_id User ID to delete the metadata of.
+	 *
+	 * @return void
+	 */
+	public function delete_meta( $user_id ) {
+		try {
+			$indexable = $this->get_indexable( $user_id, false );
+			$indexable->delete();
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+	}
+
+	/**
+	 * Saves user meta.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return void
+	 */
+	public function save_meta( $user_id ) {
+		try {
+			$indexable = $this->get_indexable( $user_id );
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+
+		$indexable->permalink = $this->get_permalink( $user_id );
+
+		$formatter      = new Indexable_Author_Formatter( $user_id );
+		$formatted_data = $formatter->format();
```

Comment:
The format function should take an indexable as it's argument and return the indexable ( or a clone ) with formatted values assigned.